src/mpi/group/grouputil.c

```diff
@@ -332,7 +332,6 @@ int MPIR_Group_check_subset(MPIR_Group * group_ptr, MPIR_Comm * comm_ptr)
         /* FIXME: MPID_Comm_get_lpid to be removed */
         uint64_t dev_lpid;
         MPID_Comm_get_lpid(comm_ptr, i, &dev_lpid, FALSE);
-        MPIR_Assert((dev_lpid >> 32) == 0);
```

Comment:
assertion was added in [37bf8320a76e57f73155b109082db6c7ccb5b956] so was presumably there to verify the lpid was not negative or something?